src/plugins/BehaviorSpecGenerator/BehaviorSpecGenerator.js

```diff
@@ -232,18 +234,29 @@ define([
             componentTypeNames = {},
             name,
             nodePath,
+            totalguardNames = {},
+            guardNames = {},
+            guardExpressions = [],
+            stateswithvalidTransitions = {},
+            totalStateNames = {},
             node;
 
         for (nodePath in nodes) {
-            var stateNames = {};
+            guardNames = {};
+            guardExpressions = [];
+            stateswithvalidTransitions = {};
+            totalStateNames = {};
+            var transitionNames = {};
             node = nodes[nodePath];
             name = this.core.getAttribute(node, 'name');
+
+
             // TODO: check all expected types and more constraints.
             if (this.isMetaTypeOf(node, this.META.ComponentType)) {
                 // This will be a java class - no special characters etc.
                 // The example is incomplete and also allows leading numbers, try at https://regex101.com/
                 //if (/^[0-9a-zA-Z_]+$/.test(name) === false) {
-                if (/^(?!abstract|continue|for|new|switch|assert|default|goto|package|synchronized|boolean|do|if|private|this|break|double|implements|protected|throw|byte|else|import|public|throws|case|enum|intanceof|return|transient|catch|extends|int|short|try|char|final|interface|static|void|class|finally|long|strictfp|volatile|const|float|native|super|while|Abstract|Continue|For|New|Switch|Assert|Default|Goto|Package|Synchronized|Boolean|Do|If|Private|This|Break|Double|Implements|Protected|Throw|Byte|Else|Import|Public|Throws|Case|Enum|Intanceof|Return|Transient|Catch|Extends|Int|Short|Try|Char|Final|Interface|Static|Void|Class|Finally|Long|Strictfp|Volatile|Const|Float|Native|Super|While)[A-Z][0-9a-z]+$/.test(name) === false) {
+                if (/^(?!abstract|continue|for|new|switch|assert|default|goto|package|synchronized|boolean|do|if|private|this|break|double|implements|protected|throw|byte|else|import|public|throws|case|enum|intanceof|return|transient|catch|extends|int|short|try|char|final|interface|static|void|class|finally|long|strictfp|volatile|const|float|native|super|while|Abstract|Continue|For|New|Switch|Assert|Default|Goto|Package|Synchronized|Boolean|Do|If|Private|This|Break|Double|Implements|Protected|Throw|Byte|Else|Import|Public|Throws|Case|Enum|Intanceof|Return|Transient|Catch|Extends|Int|Short|Try|Char|Final|Interface|Static|Void|Class|Finally|Long|Strictfp|Volatile|Const|Float|Native|Super|While)[A-Z][0-9a-zA-Z]+$/.test(name) === false) {
```

Comment:
How do I refer this variable in my file ?

I tried accessing it , gives me error "Undefined Reference"